testhelpers/testhelpers.go

```diff
@@ -391,3 +399,133 @@ func checkResponseError(r io.Reader) error {
 
 	return nil
 }
+
+func CreateSingleFileTar(path, txt string) (io.Reader, error) {
+	var buf bytes.Buffer
+	tw := tar.NewWriter(&buf)
+	if err := tw.WriteHeader(&tar.Header{Name: path, Size: int64(len(txt)), Mode: 0644}); err != nil {
+		return nil, err
+	}
+	if _, err := tw.Write([]byte(txt)); err != nil {
+		return nil, err
+	}
+	if err := tw.Close(); err != nil {
+		return nil, err
+	}
+	return bytes.NewReader(buf.Bytes()), nil
+}
+
+func RandomLayer(t *testing.T, tmpDir string) (path string, sha string, contents []byte) {
+	t.Helper()
+
+	r, err := CreateSingleFileTar("/some-file", RandString(10))
+	AssertNil(t, err)
+
+	path = filepath.Join(tmpDir, RandString(10)+".tar")
+	fh, err := os.Create(path)
+	AssertNil(t, err)
+	defer fh.Close()
+
+	hasher := sha256.New()
+	var contentsBuf bytes.Buffer
+	mw := io.MultiWriter(hasher, fh, &contentsBuf)
+
+	_, err = io.Copy(mw, r)
+	AssertNil(t, err)
+
+	sha = hex.EncodeToString(hasher.Sum(make([]byte, 0, hasher.Size())))
+
+	return path, "sha256:" + sha, contentsBuf.Bytes()
+}
+
+func CreateRemoteImage(t *testing.T) v1.Image {
```

Comment:
```suggestion
func RunnableRemoteImage(t *testing.T) v1.Image {
```
We might change `RunnableBaseImage` to `RunnableBaseImageReference` to associate these methods